backend/src/middlewares/validate-schema/validate-schema.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { HttpCode } from "shared/common/enums";
+import { AnySchema, ValidationError } from 'yup';
+import { Request, Response, NextFunction } from 'express';
+
+const validateSchema = <T extends AnySchema>(schema: T) => async (req: Request, res: Response, next: NextFunction): Promise<Response | void> => {
+  const { body } = req;
+
+  try {
+    await schema.validate(body, {
+      abortEarly: false,
+    });
+  } catch (err) {
+    if(err instanceof ValidationError) {
```

Comment:
```suggestion
    if(err instanceof ValidationError) {
```
You need to import it from package